test/monitor.js

```diff
@@ -76,62 +76,133 @@ describe('Monitor', () => {
 
     describe('start()', () => {
 
-        it('calls the init methods of all the reporters', (done) => {
+        it('correctly passes dynamic arguments to stream constructors', (done) => {
 
-            const one = new GoodReporter();
-            const two = new GoodReporter();
-            let hitCount = 0;
+            const Inc = GoodReporter.Incrementer;
+            GoodReporter.Incrementer = function (starting, multiple) {
 
-            one.init = (stream, emitter, callback) => {
+                GoodReporter.Incrementer = Inc;
+                expect(starting).to.equal(10);
+                expect(multiple).to.equal(5);
 
-                hitCount++;
-                expect(emitter.on).to.exist();
-                return callback(null);
+                return new Inc(starting, multiple);
             };
 
-            two.init = (stream, emitter, callback) => {
+            const Strng = Stringify;
```

Comment:
Is this supposed to be `Strng` to avoid conflicting with `String`? Could you just call it lowercase stringify?